### ISSUE TITLE
Initial version TODO

### ISSUE THREAD

**Issue by transitive-bullshit** [transitive-bullshit/internet-diet#1]
This issue is a lightweight way to track tasks during initial development. I plan to close it once we publish an initial release in favor of normal tagged issues.

### High priority

- [x] initial prototype
- [x] handle page updates and slow client-side refresh
- [x] add popup UI
- [x] add badge UI
- [x] add styles to default blocked page
- [x] add icon and design
- [x] make block rules customizable
- [x] add UX for adding block link rules
- [x] dedupe block rules
- [x] track the number of links and items blocked
- [x] add block this page button in popup
- [x] add block this site button in popup
- [x] add project linting
- [x] add confirmation dialogs for blocking pages and sites
- [x] add support for pausing / resuming
- [x] add support for all websites
- [x] fix popup confirmation modal overflow
- [x] inject content script dynamically
- [x] make link selection and blocking logic robust across sites
- [x] more cross-site testing
- [x] fix bug with deduping block rules resulting in incorrect createdAt times
- [x] add example gifs
- [x] add options page
  - [x] view / edit block rules
  - [x] make block page customizable
  - [x] make block effect customizable (blur vs removal)
  - [x] view total stats
  - [x] pause / unpause
  - [ ] add new block rules
- [x] improve options table (filter, search)
- [x] add hotkey command for blocking a link
- [x] add hotkey command for toggling blocking
- [x] add regression test suite for core element selection with fixtures across different sites
- [x] prep for chrome store submission
- [x] submit to chrome store
- [x] publish to chrome store
- [ ] update and test blocked item handling
- [ ] add new user onboarding UX
- [ ] add hint about how to use if opening the popup on an unrecognized site

### Low priority

- [ ] add support FAQ
- [ ] add basic website
- [ ] undo or ability to select specific item to unblock on page
  - [ ] add inline tooltip on blocked items for context and pausing
- [ ] handle sites which rely heavily on query params (like youtube)
- [ ] refactor to a single store
- [ ] create abstraction for integrating a store with a react component
- [ ] refactor differences between handling of links vs items?
- [ ] add context menu integration
- [ ] add support for firefox
- [ ] add support for safari
- [ ] linkify hostname in popup

### Known Issues

- [ ] related domains aren't handled for link selection (like twitter.com and it's use of t.co)
- [ ] implicit redirects aren't handled for link blocking and link selection
  - ex: `https://www.doordash.com/store/942804` gets expanded to `https://www.doordash.com/store/country-house-diner-clinton-hill-942804/`
  - popup isn't aware of the change
  - references to one don't block references to the other
  - doordash and caviar both have the same build
